polkadot/node/network/statement-distribution/src/v2/mod.rs

```diff
@@ -3174,6 +3181,62 @@ pub(crate) async fn handle_response<Context>(
 					"Successfully received candidate"
 				);
 
+				if !per_session.candidate_receipt_v2_enabled() &&
+					candidate.descriptor.version() == CandidateDescriptorVersion::V2
+				{
+					gum::debug!(
+						target: LOG_TARGET,
+						?candidate_hash,
+						?peer,
+						"Version 2 candidate receipts are not enabled by the runtime"
+					);
+
+					// Punish peer.
+					modify_reputation(
+						reputation,
+						ctx.sender(),
+						peer,
+						COST_INVALID_DESCRIPTOR_VERSION,
+					)
+					.await;
+					return
+				}
+
+				// Get the latest session index & check candidate descriptor session index.
```

Comment:
shouldn't we check against the session index at the relay parent?
I think it could be different from the latest seen session index in case we do a chain reversion on a fork